### PR TITLE
BE-038 (Nikolay) Add update (Vendor & Category)

### PR DESCRIPTION
**Pull request by mblack666**

### CODE REVIEW

File: src/main/java/com/exadel/discount/service/impl/VendorServiceImpl.java [hunk 1]
```diff
@@ -32,6 +33,20 @@ public VendorDTO save(CreateVendorDTO vendorDTO) {
...
+        return vendorRepository.findById(id)
+                .map( vendor -> {
+                    vendor = vendorMapper.update(vendorDTO, vendor);
+                    vendor.setId(id);
+                    VendorDTO updatedVendorDTO = vendorMapper.getDTO(vendorRepository.save(vendor));
+                    log.debug(String.format("Successfully update Vendor with ID %s", id));
+                    return updatedVendorDTO;
+                }).orElseThrow( () -> new NotFoundException(String.format("Vendor with ID %s not found", id)));
+    }
+
```
Comment (Hanna-Kalashnyk) [inline src/main/java/com/exadel/discount/service/impl/VendorServiceImpl.java]: I believe it works well, but sure could be more readable...

**Review by Hanna-Kalashnyk — DISMISSED** [review]

File: src/main/java/com/exadel/discount/repository/CategoryRepository.java [hunk 2]
```diff
@@ -10,6 +10,9 @@
 
 @Repository
 public interface CategoryRepository extends JpaRepository<Category, UUID> {
+
     @EntityGraph(attributePaths = {"discounts"})
     Optional<Category> findById(UUID uuid);
+
+    Optional<Category> findCategoryById(UUID id);
```
Comment (SergeyKoval) [inline src/main/java/com/exadel/discount/repository/CategoryRepository.java]: What is the difference between findById and findCategoryById methods?

Comment (mblack666) [reply src/main/java/com/exadel/discount/repository/CategoryRepository.java]: findById load entity category with all its discounts(its need for sorting)
findCategoryById load only entity category
I don't need all discounts, to will be edit category

**Review by Hanna-Kalashnyk — DISMISSED** [review]

File: src/main/java/com/exadel/discount/service/impl/CategoryServiceImpl.java [hunk 3]
```diff
@@ -30,6 +31,20 @@ public CategoryDTO save(CategoryDTO categoryDTO) {
         return category;
     }
 
+    @Override
+    @Transactional
+    public CategoryDTO updateCategoryById(CategoryDTO categoryDTO, UUID id) {
+        log.debug(String.format("Update Category with ID %s", id));
+        if (categoryRepository.existsById(id)) {
```
Comment (SergeyKoval) [inline src/main/java/com/exadel/discount/service/impl/CategoryServiceImpl.java]: refactor it in next way (skipping else): 
```
if (!categoryRepository.existsById(id)) {
    throw new NotFoundException(String.format("Category with ID %s not found", id));
}
....
```

**Review by SergeyKoval — DISMISSED** [review]

**Review by Hanna-Kalashnyk — APPROVED** [review]

**Review by SergeyKoval — APPROVED** [review]

**Pull request merged**